packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/DiagramRenderer.tsx

```diff
@@ -181,7 +181,9 @@ export const DiagramRenderer = ({ diagram, selection, setSelection }: DiagramRen
   };
 
   const handleEdgeClick = (event: React.MouseEvent<Element, MouseEvent>) => {
-    onEdgeClick(event);
+    if (event.target instanceof SVGElement && event.target.classList.contains('react-flow__edge-interaction')) {
```

Comment:
- That feels like a very very specific detail to hardcode
- This kind of logic does not belong in the DiagramRenderer anyway